pkg/materialize/source_webhook.go

```diff
@@ -0,0 +1,82 @@
+package materialize
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/jmoiron/sqlx"
+)
+
+type SourceWebhookBuilder struct {
+	Source
+	clusterName     string
+	size            string // TODO: size is not supported for webhook sources
+	bodyFormat      string
+	includeHeaders  bool
+	checkOptions    []string
```

Comment:
Should this be a custom struct like
```
type checkOptions struct {
    field string
    alias string
}
```
Then in the resource we can put in validation to ensure they are only using a field of type `BODY`, `HEADER`, or `SECRET`

Reply:
Just updated this, but also realized that it might be a bit more complex than my current implementation as I'll have to add a check for the SECRET as well to make sure that an actual secret is provided